src/diff/index.js

```diff
@@ -1,11 +1,40 @@
 import { EMPTY_OBJ, EMPTY_ARR } from '../constants';
-import { Component } from '../component';
+import { Component, getDomSibling } from '../component';
 import { Fragment } from '../create-element';
 import { diffChildren } from './children';
 import { diffProps, setProperty } from './props';
 import { assign, removeNode } from '../util';
 import options from '../options';
 
+const reorderChildren = newVNode => {
+	let lastVNodeChild;
+	for (let tmp = 0; tmp < newVNode._children.length; tmp++) {
+		if (newVNode._children[tmp]) {
+			newVNode._children[tmp]._parent = newVNode;
+			lastVNodeChild = newVNode._children[tmp];
+		}
+	}
+
+	if (
+		lastVNodeChild &&
+		lastVNodeChild.type &&
+		lastVNodeChild._dom &&
+		lastVNodeChild._dom.isConnected
+	) {
+		newVNode._nextDom = getDomSibling(lastVNodeChild);
+		if (newVNode._nextDom !== newVNode._dom) {
+			newVNode._children.forEach(vnode => {
+				if (typeof vnode.type == 'function') {
+					reorderChildren(vnode);
+					if (vnode._dom !== vnode._nextDom) {
+						// TODO: we should now correctly change the ordering for this node.
```

Comment:
Here we should look if the `lastVNodeChild` is still accurate, due to `children` ordering, ... This is what's tested [here](https://github.com/preactjs/preact/pull/2551/files#diff-cb37cc3064015f47cc4563ffd3b50aa9R768) this should happen deeply. With the least amount of effort